### PR TITLE
deploy: add vault creation to rbd driver deployment

### PR DESCRIPTION
**Pull request by Yuggupta27**
Currently, the script does not deploy the driver singlehandedly;
As the vault creation needs to be done prior to that.
The script now includes the vault creation so that
one script can be sufficient to deploy the rbd driver.

Fixes: #1288

### CODE REVIEW

File: examples/rbd/plugin-deploy.sh [hunk 1]
```diff
@@ -13,3 +15,15 @@ objects=(csi-provisioner-rbac csi-nodeplugin-rbac csi-config-map csi-rbdplugin-p
...
+	kms_base="../../../examples/kms/vault"
+fi
+
+cd "$kms_base" || exit 1
+
+objects=(vault csi-vaulttokenreview-rbac kms-config)
+
+for obj in "${objects[@]}"; do
+	kubectl create -f "./$obj.yaml"
+done
```
Comment (nixpanic) [inline examples/rbd/plugin-deploy.sh:29]: if you review your own PR in GitHub, you can see a warning here: "No newline at end of file"

File: examples/rbd/plugin-deploy.sh [hunk 2]
```diff
@@ -13,3 +15,15 @@ objects=(csi-provisioner-rbac csi-nodeplugin-rbac csi-config-map csi-rbdplugin-p
 for obj in "${objects[@]}"; do
 	kubectl create -f "./$obj.yaml"
 done
+
+if [[ -z $kms_base ]]; then
```
Comment (nixpanic) [inline examples/rbd/plugin-deploy.sh:19]: new code is strongly encouraged to use the `${variable}` notation

Comment (Yuggupta27) [reply examples/rbd/plugin-deploy.sh:19]: Sure. I'll change that. Thanks.

File: examples/rbd/plugin-deploy.sh [hunk 3]
```diff
@@ -1,15 +1,29 @@
...
 for obj in "${objects[@]}"; do
-	kubectl create -f "./$obj.yaml"
+	kubectl create -f "./${obj}.yaml"
+done
+
+if [[ -z ${kms_base} ]]; then
+	kms_base="../../../examples/kms/vault"
+fi
+
+cd "${kms_base}" || exit 1
```
Comment (Madhu-1) [inline examples/rbd/plugin-deploy.sh:23]: use pushd and popd instead of doing cd

Comment (Yuggupta27) [reply examples/rbd/plugin-deploy.sh]: Thanks... That will definitely be better.

File: examples/rbd/plugin-deploy.sh [hunk 4]
```diff
@@ -1,15 +1,33 @@
...
+shift
+kms_base="${1}"
 
-if [[ -z $deployment_base ]]; then
+if [[ -z ${deployment_base} ]]; then
 	deployment_base="../../deploy/rbd/kubernetes"
 fi
 
-cd "$deployment_base" || exit 1
+pushd "${deployment_base}" >/dev/null 2>&1 || exit 1
```
Comment (nixpanic) [inline examples/rbd/plugin-deploy.sh:11]: please do not redirect stderr to /dev/null, only stdout

File: examples/rbd/plugin-deploy.sh [hunk 5]
```diff
@@ -1,15 +1,33 @@
 #!/bin/bash
 
 deployment_base="${1}"
+shift
+kms_base="${1}"
 
-if [[ -z $deployment_base ]]; then
+if [[ -z ${deployment_base} ]]; then
```
Comment (nixpanic) [inline examples/rbd/plugin-deploy.sh:7]: if you're changing this, please place it between `""` too.

Comment (Yuggupta27) [reply examples/rbd/plugin-deploy.sh:7]: Sure, I'll add that. Thanks

**Comment by nixpanic** [comment]
@Mergifyio refresh

**Pull request merged**